src/vcpkg/binarycaching.cpp

```diff
@@ -211,18 +213,443 @@ namespace
+                            error_msg.append_raw('\n').append(error);
+                        msg_sink.println_error(error_msg);
+                    }
+                }
+            }
+            else
+            {
+                maybe_settings.emplace();
+                maybe_settings.get()->delete_policy = FolderSettings::DeletePolicy::OldestModificationDate;
+                const auto write_settings_file = [&]() {
+                    auto obj = FolderSettingsDeserializer::serialize(*maybe_settings.get());
+                    obj.insert("$schema",
+                               "https://raw.githubusercontent.com/microsoft/vcpkg-tool/main/docs/"
+                               "file-cache-settings.schema.json");
+                    obj.sort_keys();
+                    std::error_code ec;
+                    m_fs.create_directories(archives_root_dir, IgnoreErrors{});
+                    m_fs.write_contents(settings_path, Json::stringify(obj), ec);
+                    if (ec)
+                    {
+                        msg::println_error(msgFailedToWriteFile, msg::path = settings_path, msg::error_msg = ec);
+                    }
+                    return ec;
+                };
+                if (!write_settings_file())
+                {
+                    using namespace std::chrono;
+                    auto old_access_time = m_fs.last_access_time_now() - duration_cast<nanoseconds>(30 * 24h).count();
+                    m_fs.last_access_time(settings_path, old_access_time, VCPKG_LINE_INFO);
+                    m_fs.read_contents(settings_path, VCPKG_LINE_INFO);
+                    if (m_fs.last_access_time(settings_path, VCPKG_LINE_INFO) > old_access_time)
+                    {
+                        maybe_settings.get()->delete_policy = FolderSettings::DeletePolicy::OldestAccessDate;
+                        write_settings_file();
+                    }
+                }
+            }
+            return maybe_settings;
+        }
+
+        WriteFilePointer get_own_sync_file(const Path& sync_root_dir)
+        {
+            while (true)
+            {
+                Path path = sync_root_dir / fmt::format("{}", rand());
+                std::error_code ec;
+                WriteFilePointer wp(path, Append::NO, Overwrite::NO, ec);
+                if (!ec)
+                {
+                    return wp;
+                }
+            }
+        }
+
         size_t push_success(const BinaryPackageWriteInfo& request, MessageSink& msg_sink) override
         {
             const auto& zip_path = request.zip_path.value_or_exit(VCPKG_LINE_INFO);
             const auto archive_subpath = files_archive_subpath(request.package_abi);
 
+            size_t index = 0;
+            uint64_t file_size = m_fs.file_size(zip_path, IgnoreErrors{});
             size_t count_stored = 0;
             for (const auto& archives_root_dir : m_dirs)
             {
+                if (index >= file_cache_data.size())
```

Comment:
This whole thing looks an awful lot like it should be its own type rather than smashed like 4 layers deep inside push_success?

Reply:
Yeah I can change this